src/TSMapEditor/UI/Windows/ScriptsWindow.cs

```diff
@@ -515,15 +581,49 @@ private void ListScripts()
         {
             lbScriptTypes.Clear();
 
-            foreach (var script in map.Scripts)
+            List<Script> sortedScripts = map.Scripts;
+
+            var shouldViewTop = false; // when filtering the scroll bar should update so we use a flag here
+            if (tbFilter.Text != string.Empty && tbFilter.Text != tbFilter.Suggestion)
             {
-                lbScriptTypes.AddItem(new XNAListBoxItem() { Text = script.Name, Tag = script });
+                sortedScripts = sortedScripts.Where(sortedScript => sortedScript.Name.Contains(tbFilter.Text, StringComparison.CurrentCultureIgnoreCase)).ToList();
```

Comment:
No need to turn this into a list since we'll still use Linq to work on it during the sort. You'll just cause an extra memory allocation and heavier syntax. Also, there is likely no need to turn this into a list at all since IIRC `IEnumerable` can be worked with using `foreach`.

Further, you're calling the variable inside the lambda `sortedScript`, which makes no sense. As a singular script instance, it's just a script.

```suggestion
                sortedScripts = sortedScripts.Where(script => script.Name.Contains(tbFilter.Text, StringComparison.CurrentCultureIgnoreCase));
```